Cargo.toml

```diff
@@ -25,8 +25,15 @@ async-trait = "0.1"
 codec = { package = "parity-scale-codec", version = "2", default-features = false, features = ["derive"] }
 parking_lot = "0.11"
 
+# dependencies used by the fuzzing binary
+tokio = { version = "1.6.1", default-features = false, optional = true }
+env_logger = { version = "0.8", default-features = false, optional = true }
```

Comment:
I hate that these are non-dev dependencies here, even if they are technically optional.

Very aggressive suggestion – we already duplicate a lot of the objects you want to use from `mock` in the dummy example. Maybe instead of changing the `testing/mock` part of the repo, create another subcrate `AlephBFT-mock` (better name welcome), which would implement the things fuzz and dummy example need (copy them from mock/dummy). It would still be duplicated (we still need these in `testing/mock` for, well, testing purposes and we cannot have circular dependencies I believe), but these getting out of sync are not really a problem.

This would fix all the problems with dependencies, as well as with the strange conditional pub statements, and no need for "fuzz" as a feature.

What do you think?